### PR TITLE
Adding nemo-run recipes for NeMo 2.0 T5 

### PR DESCRIPTION
**Pull request by huvunvidia**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/recipes/t5_11b.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+from nemo.lightning.pytorch.optim.megatron import MegatronOptimizerModule
+from nemo.utils.exp_manager import TimingCallback
+
+NAME = "t5_11b"
+
+
+@run.cli.factory(name=NAME)
+def model() -> run.Config[pl.LightningModule]:
+    """
+    Factory function to create a Llama3 8B model configuration.
```
Comment (JRD971000) [inline nemo/collections/llm/recipes/t5_11b.py:41]: please change the docstring to reflect T5 instead of Llama

File: nemo/collections/llm/recipes/t5_11b.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+    Returns:
+        run.Config[pl.LightningModule]: Configuration for the Llama3 8B model.
```
Comment (JRD971000) [inline nemo/collections/llm/recipes/t5_11b.py:44]: same here

File: nemo/collections/llm/recipes/t5_11b.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        Python API usage:
+            >>> recipe = pretrain_recipe(name="t5_11b_pretrain", num_nodes=2)
+            >>> print(recipe)
+
+    Note:
+        For more details on pre-training LLMs with NeMo, see the pre-training
+        guide in the `examples/llm/pretrain/` directory.
+    """
+
+    opt_config = OptimizerConfig(
```
Comment (JRD971000) [inline nemo/collections/llm/recipes/t5_11b.py:175]: does it not work with dist fused adam?

File: tests/collections/llm/recipes/test_t5_11b.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import nemo_run as run
```
Comment (JRD971000) [inline tests/collections/llm/recipes/test_t5_11b.py:1]: what is the purpose of these test scripts here?

**Review by JRD971000 — COMMENTED** [review]
left a few comments. thanks!

File: nemo/collections/llm/recipes/t5_11b.py (hunk 2 above)
Comment (huvunvidia) [reply nemo/collections/llm/recipes/t5_11b.py:44]: Done.

File: nemo/collections/llm/recipes/t5_11b.py (hunk 1 above)
Comment (huvunvidia) [reply nemo/collections/llm/recipes/t5_11b.py:41]: Done.

File: nemo/collections/llm/recipes/t5_11b.py (hunk 3 above)
Comment (huvunvidia) [reply nemo/collections/llm/recipes/t5_11b.py:175]: Done.

File: nemo/collections/llm/recipes/t5_11b.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+    return trainer
+
+
+@run.cli.factory(target=pretrain, name=NAME)
+def pretrain_recipe(
+    dir: Optional[str] = None, name: str = "default", num_nodes: int = 20, num_gpus_per_node: int = 8, fn=pretrain
+) -> run.Partial:
+    """
+    Create a pre-training recipe for T5 220m model.
```
[the remark this answers is not in the document]
Comment (huvunvidia) [reply nemo/collections/llm/recipes/t5_11b.py:146]: Done.

**Review by JRD971000 — DISMISSED** [review]

**Review by JRD971000 — APPROVED** [review]

**Comment by JRD971000** [comment]
LGTM
